src/state/slices/tradeQuoteSlice/tradeQuoteSlice.ts

```diff
@@ -13,18 +14,22 @@ import {
 
 export type TradeQuoteSliceState = {
   activeStep: number | undefined // Make sure to actively check for undefined vs. falsy here. 0 is the first step, undefined means no active step yet
-  activeQuoteIndex: number | undefined // the selected swapper used to find the active quote in the api response
+  activeQuoteId: string | undefined // the selected quote id used to find the active quote in the api responses
   confirmedQuote: TradeQuote | undefined // the quote being executed
   tradeExecution: TradeExecutionMetadata
+  tradeQuotes: Record<string, Omit<ApiQuote, 'index'>> // mapping from quoteId to ApiQuote
```

Comment:
Ditto `ApiQuote['index']` should be `id`?
```suggestion
  tradeQuotes: Record<string, Omit<ApiQuote, 'index'>> // mapping from quoteId to ApiQuote
```

Reply:
we using index for sorting, id for selecting

Reply:
actually now you mention it, we wont need to attach the index to the quotes at all any more

Reply:
417a8fab1747aed1905dea97bf1836a9003f6b5e